broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java

```diff
@@ -7,114 +7,214 @@
  */
 package io.camunda.zeebe.broker.partitioning;
 
-import io.atomix.cluster.ClusterMembershipService;
-import io.atomix.cluster.messaging.ClusterCommunicationService;
 import io.atomix.primitive.partition.ManagedPartitionGroup;
 import io.atomix.primitive.partition.ManagedPartitionService;
 import io.atomix.primitive.partition.impl.DefaultPartitionService;
 import io.atomix.raft.partition.RaftPartitionGroup;
 import io.atomix.utils.concurrent.Futures;
 import io.camunda.zeebe.broker.PartitionListener;
+import io.camunda.zeebe.broker.clustering.ClusterServices;
+import io.camunda.zeebe.broker.exporter.repo.ExporterRepository;
 import io.camunda.zeebe.broker.partitioning.topology.TopologyManager;
 import io.camunda.zeebe.broker.partitioning.topology.TopologyManagerImpl;
 import io.camunda.zeebe.broker.partitioning.topology.TopologyPartitionListener;
-import io.camunda.zeebe.logstreams.log.LogStream;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.management.deployment.PushDeploymentRequestHandler;
+import io.camunda.zeebe.broker.system.monitoring.BrokerHealthCheckService;
+import io.camunda.zeebe.broker.system.monitoring.DiskSpaceUsageListener;
+import io.camunda.zeebe.broker.system.partitions.PartitionHealthBroadcaster;
+import io.camunda.zeebe.broker.system.partitions.ZeebePartition;
+import io.camunda.zeebe.broker.transport.commandapi.CommandApiService;
 import io.camunda.zeebe.protocol.impl.encoding.BrokerInfo;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStoreFactory;
 import io.camunda.zeebe.util.health.HealthStatus;
 import io.camunda.zeebe.util.sched.ActorSchedulingService;
-import io.camunda.zeebe.util.sched.future.ActorFuture;
-import java.util.Objects;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public final class PartitionManagerImpl
-    implements PartitionManager, TopologyManager, PartitionListener {
+public final class PartitionManagerImpl implements PartitionManager, TopologyManager {
```

Comment:
Yes we will break it down, eventually. 

My guess is that we don't need to expose `TopologyManager` interface anymore once the partition manager has its proper interface (see #7594).

Regarding the config it is a little more complicated. No, we don't need the whole config. We just need eclectic pieces of it. 

There are some low hanging fruits, but I think there are some false herrings among them.

For example:
```
return context.getBrokerCfg().getCluster().getReplicationFactor() > 1;
```

could be replaced by:
```
context.isReplicated();
```

Which IMO is reasonable to know about a partition.

Another thought one might have when looking at this code:
```
  private ActorFuture<LogStream> buildLogstream(
      final PartitionBoostrapAndTransitionContextImpl context,
      final AtomixLogStorage atomixLogStorage) {
    return LogStream.builder()
        .withLogStorage(atomixLogStorage)
        .withLogName("logstream-" + context.getRaftPartition().name())
        .withNodeId(context.getNodeId())
        .withPartitionId(context.getRaftPartition().id().id())
        .withMaxFragmentSize(context.getMaxFragmentSize())
        .withActorSchedulingService(context.getActorSchedulingService())
        .buildAsync();
  }
```

Instead of providing all the bits, we could instead provide a `Function<AtomixLogStorage, ActorFuture<LogStream>>` and then we don't need to pull all these minute dependencies into the partition in the first place.

This however also means that someone needs to curry such a function. So somebody needs to know that logstreams are a thing in a partition and how to build such a logstream.

So the trade off becomes between less dependencies on the interface between application and partition vs. not knowing or caring which bits of the configuration are relevant for the partition and its internals.

One example where I think it is desirable to just pass the config through is:
`final var databaseCfg = context.getBrokerCfg().getExperimental().getRocksdb();`

All this is a long winded way of me of saying that I have thought about this dependency, but decided to keep it as is for now. 

I guess the main decisions are whether the factory or the partition internals are responsible for interpreting the configuration. You can prepare more up front in the partition factory, but this comes at the cost that the factory needs to anticipate which bits will be needed for the partition, and how to get hold of this information.

Reply:
:+1: Thanks, well explained. I'm also not sure now if passing factories is truly the best way to go forward. I do think we can break it down though to not need the whole `BrokerCfg`, but that's something that'll probably be easier to see later once the partition manager isn't doing everything.